File: src/main/resources/output/IFC4X3_js/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcBuiltElementType/IfcWindowType/IfcWindowType.js
class IfcWindowType extends IfcBuiltElementType {
    constructor() {
        /** @type {IFCWINDOWTYPEENUM} */
        this.PredefinedType = null;
        /** @type {IFCWINDOWTYPEPARTITIONINGENUM} */
        this.PartitioningType = null;
        /** @type {IFCBOOLEAN} */
        this.ParameterTakesPrecedence = null;
        /** @type {IFCLABEL} */
        this.UserDefinedPartitioningType = null;
    }

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : (PredefinedType <> IfcWindowTypeEnum.USERDEFINED) OR ((PredefinedType = IfcWindowTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
}

File: output/ifc4x3_dev_73740fe4/entity/IfcWindowType.java
package com.topoplanet.ifc4x3_dev_73740fe4.entity;
import com.topoplanet.test.type.*;


public class IfcWindowType extends IfcBuiltElementType {

    public IfcWindowTypeEnum PredefinedType;
    public IfcWindowTypePartitioningEnum PartitioningType;
    public IfcBoolean ParameterTakesPrecedence; // OPTIONAL
    public IfcLabel UserDefinedPartitioningType; // OPTIONAL
    // INVERSE attributes:
    // CorrectPredefinedType : (PredefinedType <> IfcWindowTypeEnum.USERDEFINED) OR
    // ((PredefinedType = IfcWindowTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType));

    // WHERE constraints:
    // CorrectPredefinedType : (PredefinedType <> IfcWindowTypeEnum.USERDEFINED) OR
    // ((PredefinedType = IfcWindowTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType));
}
